lib/parse/util.rb

```diff
@@ -86,4 +88,26 @@ def Parse.object_pointer_hash(v)
       v.class_name.hash ^ v.id.hash
     end
   end
+
+
+  # NOTE: this mess is used to pass along the @client to internal objects
+  def Parse.copy_client(client, parsed_data)
```

Comment:
As you can see in the note I consider that method a problem and I would be more than happy with some refactoring and a different solution.

Thanks!